tools/clang/tools/dxcompiler/dxclinker.cpp

```diff
@@ -98,6 +140,33 @@ class DxcLinker : public IDxcLinker, public IDxcContainerEvent {
     m_pLinker.reset(DxilLinker::CreateLinker(m_Ctx, valMajor, valMinor));
   }
 
+  bool AddCompilerVersionMapEntry(LPCWSTR libName,
+                                  const hlsl::DxilCompilerVersion *pDCV,
+                                  uint32_t partSize) {
+    // Make sure it's safe to parse pDCV
+    bool valid = pDCV && partSize >= sizeof(hlsl::DxilCompilerVersion) &&
+                 partSize - sizeof(hlsl::DxilCompilerVersion) >=
+                     pDCV->VersionStringListSizeInBytes;
+    if (valid && pDCV->VersionStringListSizeInBytes) {
+        const char *vStr = (const char *)(pDCV + 1);
+        valid = vStr[pDCV->VersionStringListSizeInBytes - 1] == 0;
+    }
+
+    DXASSERT(valid, "DxilCompilerVersion part malformed");
+    if (!valid)
+        return false;
+
+    CW2A pUtf8LibName(libName, CP_UTF8);
+    std::string libNameStr = std::string(pUtf8LibName);
+    m_uniqueCompilerVersions.insert(DeserializedDxilCompilerVersion(pDCV));
```

Comment:
This returns a pair containing the iterator and a bool indicating if it was inserted. That would eliminate the need to find it in the next line.